config.go

```diff
@@ -190,7 +203,26 @@ func loadConfiguration() (*config, error) {
 		}
 	}
 
-	err := c.flags.Parse(os.Args[1:])
```

Comment:
`flags.Parse` needs to come before any code that uses values that might be set by the flags.